daemon/graphdriver/devmapper/deviceset.go

```diff
@@ -2258,6 +2291,37 @@ func (devices *DeviceSet) exportDeviceMetadata(hash string) (*deviceMetadata, er
 	return metadata, nil
 }
 
+func xfsSupported() bool {
+	// Return true only if kernel supports xfs and mkfs.xfs is available
+	exec.Command("modprobe", "xfs").Run()
+
+	f, err := os.Open("/proc/filesystems")
+	if err != nil {
+		return false
+	}
+	defer f.Close()
+
+	s := bufio.NewScanner(f)
+	for s.Scan() {
+		if strings.Contains(s.Text(), "xfs") {
```

Comment:
Right. Otherwise selinuxfs will also be recognized as having valid xfs support. Will fix it.